### ISSUE TITLE
Label Noise, Speech Noise, Cut Speech in the dataset

### ISSUE THREAD

**Issue by shangeth** [skit-ai/speech-to-intent-dataset#1]
I was doing [Dataset Cartography](https://arxiv.org/abs/2009.10795) analysis on the training dataset for the e2e SLU model based on a whisper encoder. This analysis splits the dataset into 3 parts: easy, hard, and ambiguous samples. 
![datamaps](https://user-images.githubusercontent.com/21142903/203728603-823f996b-313d-47f7-a700-feed77b7e16f.png)

After the split, I tried to analyze the hard samples to understand why these samples are harder for the model to learn. When listening to these audio samples, I found a few samples were mislabelled, a few had no speech only noise and a few sample speeches were cut in between. This analysis was only done on the train set, this has to be done on test set too.

**Comment by shangeth** [comment]
This analysis was done on the test set too and found similar errors in the test set.
![datamaps_test](https://user-images.githubusercontent.com/21142903/203748411-dcf17f08-f09d-44b9-bcaf-d492af50ae9c.png)

There are audio files like the below plot 
<img width="547" alt="Screenshot 2022-11-24 at 3 04 08 PM" src="https://user-images.githubusercontent.com/21142903/203748796-6ef9b8a5-7ff0-42ca-ba3f-d986dc196713.png">

**Comment by shangeth** [comment]
@janaab11

**Comment by shangeth** [comment]
Also adding data-maps analysis for each speaker, to find which speaker has more errors in the audio.
![datamaps-speaker](https://user-images.githubusercontent.com/21142903/204241844-1ed3d8d5-2299-4eaa-bf3c-f61f1219ec8b.png)


We can observe, speaker 5, and 1 has many hard-samples which may be incorrect and 4,7 have a few.
